wrappers/python/nntile/nntile_core/tensor.pyi

```diff
@@ -0,0 +1,55 @@
+from collections.abc import Buffer
+from typing import Sequence
+
+from .tile import Tile
+
+class TensorTraits:
+    """Integer arithmetics for tensors."""
+
+    def __init__(self, shape: Sequence[int], basetile_shape: Sequence[int]):
+        ...
+
+    def __repr__(self) -> str: ...
+
+    @property
+    def basetile_shape(self) -> tuple[int, ...]: ...
+
+    @property
+    def grid(self) -> Tile: ...
+
+    def get_tile_shape(self) -> tuple[int, ...]: ...
+    def get_grid_shape(self) -> tuple[int, ...]: ...
+
+
+class Tensor(TensorTraits):
+    """Many-dimensional tensor, presented by a set of subtensors (tiles)."""
+
+    def __init__(self, traits: TensorTraits, distribution: Sequence[int, ...],
+                 last_tag: int): ...
+
+    @property
+    def next_tag(self) -> int: ...
+
+    def from_array(self, array: Buffer) -> None: ...
```

Comment:
Actual data type is not presented here neither for Tensor nor for Buffer. How does it work in such cases? Do we have to store additional runtime member "dtype" to derive actual data types?

Reply:
Well, `collections.abc.Buffer` is dependent type but at the moment it is too much. There is a lot to do in order to `mypy` does not cry.